src/libraries/System.Security.Cryptography.Encoding/src/System.Security.Cryptography.Encoding.csproj

```diff
@@ -79,7 +85,15 @@
     <Compile Include="$(CommonPath)Microsoft\Win32\SafeHandles\X509ExtensionSafeHandles.Unix.cs"
              Link="Common\Microsoft\Win32\SafeHandles\X509ExtensionSafeHandles.Unix.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsOSX)' == 'true' or '$(TargetsMacCatalyst)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">
+    <!-- Used in Unix except Android (native crypto) and Apple -->
+  <ItemGroup Condition="'$(TargetsUnix)' == 'true' and '$(UseAndroidCrypto)' != 'true' and '$(UseAppleCrypto)' != 'true'">
+    <Compile Include="Internal\Cryptography\OidLookup.Unix.OpenSSL.cs" />
+  </ItemGroup>
+   <!-- Used in Android (native crypto) -->
+  <ItemGroup Condition="'$(UseAndroidCrypto)' == 'true'">
+    <Compile Include="Internal\Cryptography\OidLookup.Unix.PlatformNative.cs" />
```

Comment:
PlatformNative is a weird name for this, since it's the "I don't know how to ask the platform native library for this" implementation.  `OidLookup.NoFallback.cs` seems more appropriate, and not renaming `OidLookup.Unix.cs` (since that's still the one used in the "Unix" build of the library)